### ISSUE TITLE
Odoo 11: [Critical Bug] While doing Manual Reconciliation which has partial Reco.

### ISSUE THREAD

**Issue by deepa4lok** [odoo/odoo#29497]
Impacted versions:
11.0

Steps to reproduce:
Raised 2 Invoices & 1 payment for a Customer say for example

Dec 1 - Inv-001 - $40000.00
Dec 2 - Inv-002 - $20000.00
Dec 3 - Pay-001 - $50000.00 (payment received - created payment record directly)

Now I did  Reconciliation, from the menu "Manual Reconciliation", in the following order
Inv-001 40000.00 |
Inv-002 20000.00 |
pay-001      -         | 50000.00

Clicked on "Reconcile" 
Inv-001 -- Completely  Paid -- state was changed to Paid - Ok
Inv-002 -- Partiall Paid -- state is still 'Open' & Due amt is 10000 - Ok

Current behavior:--
So far good, now when I take Partner Ledger Report (_Note: Reconciled flag was False_)
I was surprised to see the when both Inv-001 also coming as O/S -- upon investigation i noticed in its associated MoveLine fully-reconciled was not set  which is bug 

So in short, while doing Manual Reco, if the last record happens to be a partial one, then the entire set of records are not set to 'Fully Reconciled' which is critical bug creating haywire with mismatched O/S amount during report generation.

Expected behavior:
Inv-001  to be fully reconciled both interms of technical & functional aspect 

Video/Screenshot link (optional):

**Comment by pedrobaeza** [comment]
Hi @deepa4lok, it stays opened because the operation is still opened as you haven't put a counterpart for the remaining 10k. Use bank statements instead of using manual reconciliation, that is intended for exact matchings, an then you will see that you can put the 20k as partial payment, and everything will be properly reconciled.

Closing as explained above.

**Comment by deepa4lok** [comment]
@pedrobaeza I think you didnt the concept right, its not about counter part -- neither am I create any JE, please follow the steps as exactly the manner I have stated, during Manual Reconciliation to reproduce this issue.
FYI -- remaining 10,000/- is considered as OPEN while doing reco. (partial o/s amount of that Invoice)

**Comment by deepa4lok** [comment]
@pedrobaeza  @Yenthe666  P.S. most of the reported bugs by me are closed w/o fixing/solution, am an Odoo enthusiast and hold a good command in both Technical & Functional terms of Odoo (since Version 5.0), All issues which I had posted are valid ones, which I do it for the best interest of Odoo's Apps, but instead of considering it, has been coolly marked as Closed, surely a blind ignorant attitude.

**Comment by pedrobaeza** [comment]
@deepa4lok Please do it the way I have told you and you won't have that problem.

**Comment by gdgellatly** [comment]
@pedrobaeza It is clearly a bug. Just because the result can be achieved another way, doesn't make it not one.  

Instructing someone that if they wish for reconciliation to work as expected they cannot use the menu titled manual reconciliation because it doesn't work - do you not see the issue? 

Why is there a menu entry if it isn't meant to work and users should only use bank statements?

**Comment by pedrobaeza** [comment]
I have explained why I do think it's not an issue, but if several of you think the other way, I don't have problem in reopening this and let Odoo itself to qualify this.

**Comment by deepa4lok** [comment]
@pedrobaeza Maybe you didnt understand the Issue here or the concept of how "Manual Reconciliation" is expected to Work.

let me put it this way -- **"Manual Reconciliation" fails w.r.t any involvement of partial Invoice/Payment**, i.e it works well only if Receivables == Payments (Which will not be the case in ideal world, especially dealing with B2B Customers)

I have a feeling I can also reproduce this Issue w.r.t BankStatement as well.

**Comment by pedrobaeza** [comment]
@deepa4lok OK, thanks for insisting in doing me to understand, and take into account that this is not something against you in particular, but to prioritize things that blocks the normal use of the system. 

As said on my initial answer, the problem is to not putting a counterpart, as manual reconciliation is expecting to have full amounts or to move the remaining part to a counterpart, but not to leave something half paid. As so, I would consider that more an improvement than a bug.

Let's see what Odoo experts tell us about this.

**Comment by fclementic2c** [comment]
I reproduced the case but it behaves as expected: https://drive.google.com/file/d/12M2Zt4iktOnndS9shxvLR2yejOinCiaS/view?usp=sharing

From the receivable report you clearly see the partial amounts as expected.
From the partner ledger, you can see the 2 invoices + the payment as long as it is a partial reconcile.
The partner ledger only remove full reconciled entries. This is the way odoo behaves and it is not a bug.

The king of presentation you are looking for is in the Open items report but this is an OCA module (https://github.com/OCA/account-financial-reporting/tree/11.0/account_financial_report).

**Comment by pedrobaeza** [comment]
@fclementic2c thanks for also investigating on this. The flow you have followed is not the same as said by @deepa4lok, and the "Manual reconciliation" entry is there in Odoo core without any OCA module, but indeed there are multiple forms for achieving this without problems, so no "Critical bug" as the title says, and that's why I initially close the issue.

**Comment by pedrobaeza** [comment]
@william-andre thanks for confirming my initial suspiction.

**Comment by deepa4lok** [comment]
@william-andre  Logically you are right, but then when system is unable to do, it should at least enforce user to select writeoff or any account, but it didn't, it just simply allowed reco without any hassle but with silent internal bugs, which is apparently called as **a bug**  So you guys can save your justifications.
